src/dataset.py

```diff
@@ -1,87 +1,58 @@
+from typing import Dict, Tuple, List, Optional, Any
 
 from flask import render_template
 from sodapy import Socrata
+import sqlite3
 
 
 # Set up the sodapy client
 client = Socrata("data.seattle.gov", None)
 LICENSE_DATASET = "enxu-fgzb"
 SALARY_DATASET = "2khk-5ukd"
-BADGE_DATASET_PATH = Path(__file__).parent / "data" / "SPD_roster_11-15-20.csv"
-_DataDict = Dict[str, Dict[str, str]]
-_DataNameLookup = Dict[str, _DataDict]
 
+# Connect to SQL database
+sql_conn = sqlite3.connect('./data/1-312-data.db')
+sql_curs = sql_conn.cursor()
 
-class C:
-    """Columns"""
-
-    SERIAL = "Serial"
-    FIRST = "First"
-    MIDDLE = "Middle"
-    LAST = "Last"
-    TITLE = "Title Description"
-    UNIT = "Unit Description"
-
-    fields = [SERIAL, FIRST, MIDDLE, LAST, TITLE, UNIT]
-
-
-def _data() -> Tuple[_DataDict, _DataNameLookup]:
-    """Set up initial dataset"""
-    print("Loading initial badge data")
-    badges = {}
-    names = defaultdict(dict)
-    with BADGE_DATASET_PATH.open("r") as file:
-        reader = DictReader(file)
-        for row in reader:
-            r = {k: row[k] for k in C.fields}
-            badges[row[C.SERIAL]] = r
-            # There may be numerous rows per unique name, so this makes them
-            # easiest to index appropriately/overwrite with the most current value
-            names[row[C.LAST]][row[C.SERIAL]] = r
-
-    return badges, names
 
+def license_lookup(license: str) -> str:
+    if license:
+        try:
+            results = client.get(
+                LICENSE_DATASET, limit=1, where=f"license='{license.upper()}'"
+            )
+            if not results:
+                html = "<p><b>No vehicle found for this license in public dataset</b></p><p>(not all undercover vehicles have available information)</p>"
+            else:
+                r = results[0]
+                html = render_template("license.j2", **r)
+
+        except Exception as err:
+            print(f"Error: {err}")
+            html = f"<p><b>Error:</b> {err}"
+
+        print("Final HTML:\n{}".format(html))
+        return html
 
-BADGE_DATASET, NAME_DATASET = _data()
+    else:
+        return ""
 
 
-def license_lookup(license: str) -> str:
-    try:
-        results = client.get(
-            LICENSE_DATASET, limit=1, where=f"license='{license.upper()}'"
-        )
-        if not results:
-            html = "<p><b>No vehicle found for this license in public dataset</b></p><p>(not all undercover vehicles have available information)</p>"
-        else:
-            r = results[0]
-            html = render_template("license.j2", **r)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
-
-    print("Final HTML:\n{}".format(html))
-    return html
-
-
-def _augment_with_salary(record: Dict[str, str]) -> Dict[str, str]:
-    first = record[C.FIRST]
-    last = record[C.LAST]
-    middle = record[C.MIDDLE]
+def _augment_with_salary(record: Tuple) -> Dict[str, str]:
```

Comment:
With the change suggested to the function below, this would be `record: RosterRecord`, and the values could be accessed as attributes rather than tuple indices.